lib/mix/tasks/phx.gen.html.ex

```diff
@@ -2,54 +2,66 @@ defmodule Mix.Tasks.Phx.Gen.Html do
-
-  Overall, this generator will add the following files to `lib/`:
-
-    * a context module in `lib/app/accounts.ex` for the accounts API
-    * a schema in `lib/app/accounts/user.ex`, with an `users` table
-    * a controller in `lib/app_web/controllers/user_controller.ex`
-    * an HTML view collocated with the controller in `lib/app_web/controllers/user_html.ex`
-    * default CRUD templates in `lib/app_web/controllers/user_html`
-
-  ## The context app
-
-  A migration file for the repository and test files for the context and
-  controller features will also be generated.
-
-  The location of the web files (controllers, HTML views, templates, etc) in an
-  umbrella application will vary based on the `:context_app` config located
-  in your applications `:generators` configuration. When set, the Phoenix
-  generators will generate web files directly in your lib and test folders
-  since the application is assumed to be isolated to web specific functionality.
-  If `:context_app` is not set, the generators will place web related lib
-  and test files in a `web/` directory since the application is assumed
-  to be handling both web and domain specific functionality.
-  Example configuration:
-
-      config :my_app_web, :generators, context_app: :my_app
+  > over distinct contexts (e.g., `Accounts.User` and `Payments.User`).
+
+  This generator adds the following files to `lib/`:
+
+    * a controller in `lib/my_app_web/controllers/user_controller.ex`
+    * default CRUD HTML templates in `lib/my_app_web/controllers/user_html`
+    * an HTML view collocated with the controller in `lib/my_app_web/controllers/user_html.ex`
+    * a schema in `lib/my_app/accounts/user.ex`, with an `users` table
+    * a context module in `lib/my_app/accounts.ex` for the accounts API
+  
+  Additionally, this generator creates the following files:
+    
+    * a migration for the schema in `priv/repo/migrations`
+    * a controller test module in `test/my_app/controllers/user_controller_test.exs`
+    * a context test module in `test/my_app/accounts_test.exs`
+    * a context test helper module in `test/support/fixtures/accounts_fixtures.ex` 
+  
+  If the context already exists, this generator injects functions for the given resource into
+  the context, context test, and context test helper modules.
+
+  ## Umbrella app configuration
+
+  `my_app_umbrella` assumes that `my_app_web` will handle both web and domain specific functionality. 
+  You can change the location of your controllers — with collocated views — and template files by
+  changing the value of the `:context_app` configuration option.
+  
+  Phoenix generators will generate web files directly in your lib and test folders since 
+  the application is assumed to be isolated to web specific functionality.
+  
+  Example configuration in `my_app_umbrella/config/config.exs`:
+
+    config :my_app_web,
+      ecto_repos: [Stuff.Repo],
+      generators: [context_app: :my_app]
```

Comment:
```suggestion
  By default, Phoenix injects both web and domain specific functionality into the same
  application. When using umbrella applications, those concerns are typically broken
  into separate apps. You can teach Phoenix to place your context and migration files
  into a different application via the `:context_app` configuration option.

  Example configuration in `my_app_umbrella/config/config.exs`:

    config :my_app_web,
      ecto_repos: [Stuff.Repo],
      generators: [context_app: :my_app]
```

Reply:
I found the paragraphs above to be wrong. umbrella apps generated by phoenix does not assume it all goes into the same directory. the above is an attempt to simplify it but the previous text had more details, so we probably want to keep it.